sim/src/image.rs

```diff
@@ -1796,12 +1796,13 @@ fn verify_trailer(flash: &SimMultiFlash, slot: &SlotInfo,
 
     failed |= match magic {
         Some(v) => {
-            if v == 1 && &copy[24..] != MAGIC {
+            let magic_off = (c::boot_max_align() * 3) + (c::boot_magic_sz() - MAGIC.len());
+            if v == 1 && &copy[magic_off..] != MAGIC {
```

Comment:
@gustavonihei Those changes to remove hard-coded values look good, please submit them in a separate PR, and I mean all changes to `image.rs` but the adding of `16, 32` alignments to testing.

Reply:
Sure, created PR https://github.com/mcu-tools/mcuboot/pull/1226.